src/util/PropertyPath.ts

```diff
@@ -0,0 +1,35 @@
+export type MaxPropertyPathRecursionDepth = 5;
```

Comment:
This is going to be a breaking change for users since we support TypeScript versions far behind `4` where this syntax is supported. We have to use `string` for now, but in the future we have a strategy to use this syntax everywhere.

Reply:
oh, didn't realize pre-4 was still supported :(

Reply:
yeah 0.2.x is quite old and we are trying to avoid breaking changes. I hope when we finally release `0.3.0` it will use the latest TypeScript version.

Reply:
let me know if I can be of assistance with `0.3.0` 👍